RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractZoneIdentifier.java

```diff
@@ -0,0 +1,398 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.recentactivity;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.logging.Level;
+import org.openide.util.NbBundle.Messages;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.coreutils.NetworkUtils;
+import org.sleuthkit.autopsy.ingest.DataSourceIngestModuleProgress;
+import org.sleuthkit.autopsy.ingest.IngestJobContext;
+import org.sleuthkit.autopsy.ingest.IngestServices;
+import org.sleuthkit.autopsy.ingest.ModuleDataEvent;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.ReadContentInputStream;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * Extract the <i>:Zone.Indentifier<i> alternate data stream files.
+ */
+final class ExtractZoneIdentifier extends Extract {
+
+    private static final Logger LOG = Logger.getLogger(ExtractEdge.class.getName());
+
+    private static final String ZONE_IDENIFIER_FILE = "%:Zone.Identifier"; //NON-NLS
+    private static final String ZONE_IDENIFIER = ":Zone.Identifier"; //NON-NLS
+
+    @Messages({
+        "ExtractZone_process_errMsg_find=A failure occured while searching for :Zone.Indentifier files.",
+        "ExtractZone_process_errMsg=A error occured processing ':Zone.Indentifier' files.",
+        "ExtractZone_progress_Msg=Extracting :Zone.Identifer files"
+    })
+
+    @Override
+    void process(Content dataSource, IngestJobContext context, DataSourceIngestModuleProgress progressBar) {
+
+        progressBar.progress(Bundle.ExtractZone_progress_Msg());
+
+        List<AbstractFile> zoneFiles = null;
+        try {
+            zoneFiles = findZoneFiles(dataSource);
+        } catch (TskCoreException ex) {
+            addErrorMessage(Bundle.ExtractZone_process_errMsg_find());
+            LOG.log(Level.SEVERE, "Unable to find zone identifier files, exception thrown. ", ex);
+        }
+
+        if (zoneFiles == null || zoneFiles.isEmpty()) {
+            return;
+        }
+        
+        ArrayList<Long> knownPathIDs = null;
+        try {
+            knownPathIDs = getPathIDsForType(BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_DOWNLOAD);
+        } catch (TskCoreException ex) {
+            addErrorMessage(Bundle.ExtractZone_process_errMsg());
+            LOG.log(Level.SEVERE, "Failed to build PathIDs List for TSK_WEB_DOWNLOAD", ex);
+        }
+
+        if (knownPathIDs == null) {
+            return;
```

Comment:
the only way knownPathIDs can be null is if getPathIDsForType throws an exception, so I would put the return in the try block, and skip the null check.

Reply:
I originally had it in the catch, but I thought this was a bit more explicit.   I guess moving it does save the check.